File: src/react-staff-admin/src/core/di/injection.ts
import { AuthRemoteDataSource } from '../../features/auth/data/datasources/auth.remote_datasource';
import { AuthRepositoryImpl } from '../../features/auth/data/repositories/auth.repository_impl';
import { LoginUseCase } from '../../features/auth/domain/usecases/login.usecase';

import { BookingRemoteDataSource } from '../../features/booking/data/datasources/booking.remote_datasource';
import { BookingRepositoryImpl } from '../../features/booking/data/repositories/booking.repository_impl';
import { GetBookingsUseCase } from '../../features/booking/domain/usecases/get_bookings.usecase';
import { GetBookingDetailUseCase } from '../../features/booking/domain/usecases/get_booking_detail.usecase';
import { CreateBookingUseCase } from '../../features/booking/domain/usecases/create_booking.usecase';
import { UpdateBookingStatusUseCase } from '../../features/booking/domain/usecases/update_booking_status.usecase';

import { NotificationRemoteDataSource } from '../../features/notification/data/datasources/notification.remote_datasource';
import { NotificationRepositoryImpl } from '../../features/notification/data/repositories/notification.repository_impl';
import { GetNotificationsUseCase } from '../../features/notification/domain/usecases/get_notifications.usecase';
import { MarkNotificationReadUseCase } from '../../features/notification/domain/usecases/mark_notification_read.usecase';
import { MarkAllNotificationsReadUseCase } from '../../features/notification/domain/usecases/mark_all_notifications_read.usecase';
import { CreateNotificationUseCase } from '../../features/notification/domain/usecases/create_notification.usecase';

// 1. Data Sources
const authRemoteDataSource = new AuthRemoteDataSource();
const bookingRemoteDataSource = new BookingRemoteDataSource();
const notificationRemoteDataSource = new NotificationRemoteDataSource();

// 2. Repositories
const authRepository = new AuthRepositoryImpl(authRemoteDataSource);
const bookingRepository = new BookingRepositoryImpl(bookingRemoteDataSource);
const notificationRepository = new NotificationRepositoryImpl(notificationRemoteDataSource);

// 3. Use Cases
export const loginUseCase = new LoginUseCase(authRepository);
export const getBookingsUseCase = new GetBookingsUseCase(bookingRepository);
export const getBookingDetailUseCase = new GetBookingDetailUseCase(bookingRepository);
export const createBookingUseCase = new CreateBookingUseCase(bookingRepository);
export const updateBookingStatusUseCase = new UpdateBookingStatusUseCase(bookingRepository);

export const getNotificationsUseCase = new GetNotificationsUseCase(notificationRepository);
export const markNotificationReadUseCase = new MarkNotificationReadUseCase(notificationRepository);
export const markAllNotificationsReadUseCase = new MarkAllNotificationsReadUseCase(notificationRepository);
export const createNotificationUseCase = new CreateNotificationUseCase(notificationRepository);

File: src/sports_management/modules/notification_module/lib/application/di/injection.dart
import 'package:get_it/get_it.dart';
import 'package:server_module/server_module.dart';
import '../../data/datasources/remote/notification_remote_data_source.dart';
import '../../domain/repositories/notification_repository.dart';
import '../../data/repositories/notification_repository_impl.dart';
import '../../domain/usecases/create_notification_usecase.dart';
import '../../domain/usecases/get_notifications_usecase.dart';
import '../../domain/usecases/mark_notification_read_usecase.dart';
import '../../domain/usecases/mark_all_notifications_read_usecase.dart';
import '../../presentation/cubit/notification_cubit.dart';

final sl = GetIt.instance;

Future<void> initInjection() async {
  if (!sl.isRegistered<NotificationRemoteDataSource>()) {
    sl.registerLazySingleton<NotificationRemoteDataSource>(
      () => NotificationRemoteDataSourceImpl(sl<NotificationService>()),
    );
  }

  if (!sl.isRegistered<AppNotificationRepository>()) {
    sl.registerLazySingleton<AppNotificationRepository>(
      () => AppNotificationRepositoryImpl(sl<NotificationRemoteDataSource>()),
    );
  }

  if (!sl.isRegistered<CreateNotificationUseCase>()) {
    sl.registerLazySingleton<CreateNotificationUseCase>(
      () => CreateNotificationUseCase(sl<AppNotificationRepository>()),
    );
  }

  if (!sl.isRegistered<GetNotificationsUseCase>()) {
    sl.registerLazySingleton<GetNotificationsUseCase>(
      () => GetNotificationsUseCase(sl<AppNotificationRepository>()),
    );
  }

  if (!sl.isRegistered<MarkNotificationReadUseCase>()) {
    sl.registerLazySingleton<MarkNotificationReadUseCase>(
      () => MarkNotificationReadUseCase(sl<AppNotificationRepository>()),
    );
  }

  if (!sl.isRegistered<MarkAllNotificationsReadUseCase>()) {
    sl.registerLazySingleton<MarkAllNotificationsReadUseCase>(
      () => MarkAllNotificationsReadUseCase(sl<AppNotificationRepository>()),
    );
  }

  if (!sl.isRegistered<NotificationCubit>()) {
    sl.registerLazySingleton<NotificationCubit>(
      () => NotificationCubit(
        getNotifications: sl<GetNotificationsUseCase>(),
        markRead: sl<MarkNotificationReadUseCase>(),
        markAllRead: sl<MarkAllNotificationsReadUseCase>(),
      ),
    );
  }
}
